exporters/otlp/otlplog/otlploghttp/client_test.go

```diff
@@ -700,6 +700,30 @@ func TestConfig(t *testing.T) {
 		assert.Len(t, rCh, 0, "failed HTTP responses did not occur")
 	})
 
+	t.Run("WithRetryAndExporterErr", func(t *testing.T) {
+		exporterErr := errors.New("rpc error: code = Unavailable desc = service.name not found in resource attributes")
+		rCh := make(chan exportResult, 1)
+		rCh <- exportResult{Err: &httpResponseError{
+			Status: http.StatusTooManyRequests,
+			Err:    exporterErr,
+		}}
+		exp, coll := factoryFunc("", rCh, WithRetry(RetryConfig{
+			Enabled: false,
+		}))
+		ctx := context.Background()
+		t.Cleanup(func() { require.NoError(t, coll.Shutdown(ctx)) })
+		// Push this after Shutdown so the HTTP server doesn't hang.
+		t.Cleanup(func() { close(rCh) })
+		t.Cleanup(func() { require.NoError(t, exp.Shutdown(ctx)) })
+		err := exp.Export(ctx, make([]log.Record, 1))
+		assert.ErrorContains(t, err, exporterErr.Error())
+
+		// To test the `Unwrap` and `As` function of retryable error
+		var retryErr *retryableError
+		assert.ErrorAs(t, err, &retryErr)
+		assert.ErrorIs(t, err, *retryErr)
```

Comment:
This is a no-op test. `retryErr` was asserted to be `err` on the previous line.

I would use `assert.ErrorIs(t, err, exporterErr)` in place of `ErrorContains` on line 719. This will effectively test the same thing `ErrorContains` tests (the error string is guaranteed to be equal if the identity is asserted) while also testing the unwrap functionality.

Reply:
I understand that `ErrorIs` can be used but here it will fail because `err` and `exporterErr` are different instances, unlike with `retryErr` where `ErrorAs` has passed reference of wrapped err and `ErrorIs` is true in next assert. 

`exporterErr` is created to be passed in exporter result but it is not directly wrapped internally in the flow like  **CASE 3** in this link https://go.dev/play/p/ap14MIXRk21

Here it is following **CASE 2** in the above link, that's why `ErrorIs` will fail here.

Let me know if there is a gap in this.

Reply:
Right, makes sense :+1: